This patch is needed to allow builds with newer versions of
the GNU libc (2.16+).

The upstream fix was:

  commit 66712c23388e93e5c518ebc8515140fa0c807348
  Author: Eric Blake <eblake@redhat.com>
  Date:   Thu Mar 29 13:30:41 2012 -0600

      stdio: don't assume gets any more

      Gnulib intentionally does not have a gets module, and now that C11
      and glibc have dropped it, we should be more proactive about warning
      any user on a platform that still has a declaration of this dangerous
      interface.

      * m4/stdio_h.m4 (gl_STDIO_H, gl_STDIO_H_DEFAULTS): Drop gets
      support.
      * modules/stdio (Makefile.am): Likewise.
      * lib/stdio-read.c (gets): Likewise.
      * tests/test-stdio-c++.cc: Likewise.
      * m4/warn-on-use.m4 (gl_WARN_ON_USE_PREPARE): Fix comment.
      * lib/stdio.in.h (gets): Make warning occur in more places.
      * doc/posix-functions/gets.texi (gets): Update documentation.
      Reported by Christer Solskogen.

      Signed-off-by: Eric Blake <eblake@redhat.com>

This patch just gets rid of the offending part.

--- m4-1.4.16/lib/stdio.in.h.orig	2012-09-01 01:05:36.000000000 +0200
+++ m4-1.4.16/lib/stdio.in.h	2012-09-01 01:05:42.000000000 +0200
@@ -158,12 +158,6 @@ _GL_WARN_ON_USE (fflush, "fflush is not
                  "use gnulib module fflush for portable POSIX compliance");
 #endif
 
-/* It is very rare that the developer ever has full control of stdin,
-   so any use of gets warrants an unconditional warning.  Assume it is
-   always declared, since it is required by C89.  */
-#undef gets
-_GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
-
 #if @GNULIB_FOPEN@
 # if @REPLACE_FOPEN@
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
